miden-crypto/src/dsa/rpo_falcon512/signature.rs

```diff
@@ -33,19 +33,59 @@ use crate::Word;
 /// Here h is a polynomial representing the public key and pk is its digest using the Rpo256 hash
 /// function. c is a polynomial that is the hash-to-point of the message being signed.
 ///
-/// The polynomial h is serialized as:
-/// 1. 1 byte representing the log2(512) i.e., 9.
-/// 2. 896 bytes for the public key itself.
+/// Another differentiating point from the reference implementation is the determinism in its
+/// signing process. The approach used to achieve this is the one proposed in [1]. The main
+/// challenge in making the signing procedure deterministic is ensuring that the same secret key
+/// is never used to produce two inequivalent signatures for the same `c`. For a precise definition
+/// of equivalence of signatures see [1]. The reference implementation uses a random nonce per
+/// signature in order to make sure that, with overwhelming probability, no two c-s will ever
+/// repeat and this non-repetition turns out to be enough to make the security proof of
+/// the underlying construction go through in the random-oracle model.
+/// Making the signing process deterministic means that we cannot rely on the above use of nonce
```

Comment:
nit: I would add a newline above this line.

Reply:
done